src/cata_tiles.cpp

```diff
@@ -1317,8 +1317,13 @@ bool cata_tiles::draw_from_id_string(std::string id, TILE_CATEGORY category,
             seed = g->mon_at( tripoint( x, y, z_coord ) );
             break;
     }
+    // this should preserve the deterministic nature of the game based on original seed
+    // even when a tileset changes
+    auto temp = rand();
     srand( seed );
     unsigned int loc_rand = rand();
+    srand( temp );
```

Comment:
Have you tested this? Neither `man rand` nor `man srand` (which happens to be the same page) indicate that the result of `rand` can be used to re-seed the RNG to its *previous* state. The [man page](http://linux.die.net/man/3/rand) even contains an example implementation that definitively does not behave that way.

Maybe you should use one of the random number generators from [header \<random\>](http://www.cplusplus.com/reference/random/) in `cata_tiles`?